api-get-object.go

```diff
@@ -26,6 +26,30 @@ import (
 	"time"
 )
 
+// GetSecuredObject returns a readable object
+func (c Client) GetSecuredObject(bucketName, objectName string, securedObject SecureData) (io.Reader, error) {
+
+	if securedObject == nil {
+		return nil, errors.New("cannot deal with nil secure object")
```

Comment:
Provide a typed error encapsulated as "ErrorResponse" perhaps like` ErrInvaildArgument("Unable to recognize empty encryption properties") `